agroal-pool/src/main/java/io/agroal/pool/ConnectionPool.java

```diff
@@ -403,7 +431,7 @@ private boolean performValidation(ConnectionHandler handler, ConnectionHandler.S
         fireBeforeConnectionValidation( listeners, handler );
         if ( handler.isValid() && handler.setState( VALIDATION, targetState ) ) {
             fireOnConnectionValid( listeners, handler );
-            synchronizer.releaseConditional();
+            handlerTransferQueue.tryTransfer( handler );
```

Comment:
maybe you want to offer it instead?
If a later consumer block after a "failed"  `tryTransfer` it won't find anything and will wait forever.